var/spack/packages/leveldb/package.py

```diff
@@ -0,0 +1,29 @@
+import os
+import glob
+from spack import *
+
+class Leveldb(Package):
+    """LevelDB is a fast key-value storage library written at Google
+    that provides an ordered mapping from string keys to string values."""
+
+    homepage = "https://github.com/google/leveldb"
+    url      = "https://github.com/google/leveldb/archive/v1.18.tar.gz"
+
+    version('1.18', '73770de34a2a5ab34498d2e05b2b7fa0')
+
+    depends_on("snappy")
+
+    def install(self, spec, prefix):
+        make()
+
+        mkdirp(prefix.include)
+        mkdirp(prefix.lib)
+
+        cp = which('cp')
```

Comment:
@adammoody: This is not portable (e.g., BSD `cp` on my mac doesn't have the `--preserve-links` option) but since you're the only one using this package at the moment I'm fine with merging into develop.

If you need `cp --preserve-links`, can you add something to `llnl.util.filesystem` to do it?  A lot The functions in there are already included in the build environment (see `build_environment.py` and `spack/__init__.py`), and I think preserving links would be useful.  Alternately I guess you could use `tar` to do this, but eventually I would like Spack not to rely on `tar`, either.